src/compiler/transformers/update-component-class.ts

```diff
@@ -31,25 +47,48 @@ export const updateComponentClass = (
   }
 
   // ESM with export
-  return createConstClass(transformOpts, classNode, heritageClauses, members);
+  return createConstClass(transformOpts, classNode, heritageClauses, members, moduleFile);
```

Comment:
any idea why we do this in the esm case?

Reply:
AFAIK, for "optimizations" in https://github.com/ionic-team/stencil/commit/438c6b310f24d93dc82aba1614e0f9d91dbccd7a. The issue that commit links to in the TS repo implies that classes were getting wrapped in IIFE's if they had static members - a few things aren't immediately clear to me:
1. Whether the optimization is purely from a TS perspective once that commit is applied again, or has something to do with how Rollup interacts with variable declarations 
2. What was broken in the CJS export to begin with

I believe it was attempted to do this for both CJS and ESM, but it was partially reverted in https://github.com/ionic-team/stencil/commit/4eb5b421cfdff5a73b96837656997a06180de806

Reply:
ah, optimizations...

ok sounds like we probably don't want to mess with this now, I was just wondering if you had more context on it 👍